Lib/app.rb

```diff
@@ -0,0 +1,122 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rental'
+require_relative 'classroom'
+
+class App
+  attr_accessor :books, :people, :rentals
+
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_books
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+  end
+
+  def list_people
+    @people.each { |person| puts "#{person.class}: Name: #{person.name}, Id: #{person.id}, Age: #{person.age}" }
+  end
+
+  def create_person
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_type = gets.chomp
+
+    case person_type
+    when '1'
+      student = create_student
+      @people << student
+      puts 'Student created successfully'
+    when '2'
+      teacher = create_teacher
+      @people << teacher
+      puts 'Teacher created successfully'
+    else
+      puts 'Invalid option'
+    end
+  end
+
+  def create_student
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Classroom: '
+    classroom_name = gets.chomp
+    classroom = Classroom.new(classroom_name)
+
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase == 'y'
+
+    Student.new(name, age, classroom, parent_permission: parent_permission)
+  end
+
+  def create_teacher
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    Teacher.new(name, age, specialization)
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+
+    print 'Author: '
+    author = gets.chomp
+
+    @books << Book.new(title, author)
+    puts 'Book created successfully'
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number:'
+    @books.each_with_index { |book, index| puts "#{index}) #{book.title}" }
+
+    book_index = gets.chomp.to_i
+
+    puts 'Select a person from the following list by number:'
+    @people.each_with_index { |person, index| puts "#{index}) #{person.name}" }
+
```

Comment:
- [ ] As requested by the previous reviewer in this [review link](https://github.com/hafiz1379/OOP-school-library/pull/4#discussion_r1347384601) when displaying the list of people as shown in the reference video and image below, there are some information that should be dispalyed such as the type of person, `[student] or [teacher]`, `name, ID & age`
- project requirement
![inf](https://github.com/hafiz1379/OOP-school-library/assets/89055502/9d7a7b10-562b-4493-9736-1c2da9d3f223)
- your project
![se](https://github.com/hafiz1379/OOP-school-library/assets/89055502/2a1fb940-f771-496a-8a1e-df4716f51bd3)
- [ ] Please kindly refer to the [video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) and make sure you are not missing any important information before submitting :+1: